language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/codegen/Update.java

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.javaapi.data.codegen;
+
+import com.daml.ledger.javaapi.data.Command;
+
+public abstract class Update<R> {
+  public abstract Command command();
+
+  public abstract ValueDecoder<R> returnTypeDecoder();
```

Comment:
If the `submitWithResult` utility is too difficult to do without making `Update` an ADT, then you probably only need two cases: one where we are looking for an exercise whose result to decode, in which case `ValueDecoder<R>` is defined, and one that is only a create, in which case only `Function<String, R>` is defined. And these cases should probably be package-private.

This is because, when we decode a transaction to the result of a create, _we aren't decoding a `Value`_. Instead, we are decoding a `CreatedEvent`, which just has the contract ID as a string.

Overall, I suggest making `Update` somewhat more opaque than it is right now. That will maximize our flexibility when it comes to extensions, e.g. defining "compound updates" that contain more than one `Command` (and what will that do about the `command` method here? just food for thought).